File: mpi/exercises/exercise_4/provided.f90
!> @brief This exercise is to find the bug in a simple MPI hello world.
PROGRAM main
	USE mpi

	IMPLICIT NONE

	INTEGER :: ierror
	INTEGER :: my_rank
	INTEGER :: comm_size

    CALL MPI_Comm_rank(MPI_COMM_WORLD, my_rank, ierror)

    CALL MPI_Comm_size(MPI_COMM_WORLD, comm_size, ierror)

    WRITE(*, '(A,I0,A,I0,A)') '"Hello World!" from MPI process ', my_rank ,'. We are ', comm_size ,' MPI processes.'

    CALL MPI_Finalize(ierror)
END PROGRAM main

File: mpi/exercises/exercise_4/provided.c
#include <stdio.h>
#include <mpi.h>

/**
 * @brief This exercise is to find the bug in a simple MPI hello world.
 **/
int main(int argc, char* argv[])
{
	int my_rank;
	MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);

	int comm_size;
	MPI_Comm_size(MPI_COMM_WORLD, &comm_size);

	printf("\"Hello World!\" from MPI process %d. We are %d MPI processes.\n", my_rank, comm_size);

	MPI_Finalize();

	return 0;
}